src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -345,10 +346,12 @@ trait Infer extends Checkable {
      *  by existentially bound variables.
      */
     def makeFullyDefined(tp: Type): Type = {
-      var tparams: List[Symbol] = Nil
+      var tparams_ : ListBuffer[Symbol] = null
+      def tparams: ListBuffer[Symbol] = { if (tparams_ == null) tparams_ = ListBuffer.empty ; tparams_ }
```

Comment:
This should be (manually) inlined; as it stands, it'll make an `ObjectRef` wrapping `tparams_` by which to modify the var in an outer scope.

Reply:
I've pushed a commit to this PR to move `tparams_` into a field of the `TypeMap`.